src/expr/src/scalar/func.rs

```diff
@@ -848,6 +858,19 @@ fn sub_date_interval<'a>(a: Datum<'a>, b: Datum<'a>) -> Result<Datum<'a>, EvalEr
     Ok(dt.try_into()?)
 }
 
+fn sub_date_int32<'a>(a: Datum<'a>, b: Datum<'a>) -> Result<Datum<'a>, EvalError> {
+    let date = a.unwrap_date();
+    let days = b
+        .unwrap_int32()
+        .checked_neg()
+        .ok_or(EvalError::DateOutOfRange)?;
+
+    Ok(date
+        .checked_add(days)
```

Comment:
Why not use `checked_neg` here? Is this method not available on `Date`?

Reply:
It's not available, but I suppose I should probably just add it and then use that.

Reply:
Done